spec/sane_patch_spec.rb

```diff
@@ -2,4 +2,108 @@
   it "has a version number" do
     expect(SanePatch::VERSION).not_to be nil
   end
+
+  describe ".patch" do
+    let(:gem_name) { "some_gem" }
+    let(:installed_gem_version) { nil }
+
+    before do
+      if installed_gem_version
+        allow(Gem.loaded_specs).to receive(:[]).with(gem_name).and_return(
+          double("Fake Gemspec", version: Gem::Version.new(installed_gem_version))
+        )
+      end
+    end
+
+    context "when gem is not installed" do
+      it "raises error for non-existing gem" do
+        expect do
+          ::SanePatch.patch(gem_name, "0.0.1")
+        end.to raise_error(::SanePatch::Errors::GemAbsent)
+      end
+    end
+
+    context "when gem is installed" do
```

Comment:
Maybe we could use the `:aggregate_failures` metatag on the `context` rather than having it in every spec bellow.